documentation/DesignDocumentation.md

```diff
@@ -6,24 +6,48 @@ The two main forms of design that are most relevant to the Tornelo Scoresheet ap
 
 **System Architecture**
 
-The System Architecture includes the way the model, the way the program is structured and the programming languages and tools used. 
+The System Architecture includes the way the model, the way the program is structured and the programming languages and tools used.
 
-Key areas of this aspect of the system design and rationale behind relevant design choices are documented in Section 1 of this document. 
+Key areas of this aspect of the system design and rationale behind relevant design choices are documented in Section 1 of this document.
 
 **User Interface (UI) Design**
 
-The User Interface Design is a holistic approach to deciding how the applications functional elements are displayed on the screen. This includes considering the users and their experiences, as well as the necessary features and functional requirements. 
+The User Interface Design is a holistic approach to deciding how the applications functional elements are displayed on the screen. This includes considering the users and their experiences, as well as the necessary features and functional requirements.
 
-Key areas of this aspect of the system design and rationale behind relevant design choices are documented in Section 1 of this document. 
+Key areas of this aspect of the system design and rationale behind relevant design choices are documented in Section 1 of this document.
 
 ## 1 System Architecture
-TODO
 
-### 1.1 Class Diagram and Model
-TODO
-
-### 1.2 Folder Structure
-TODO
+### 1.1 Architecture/Model Diagram
+![architecture_diagram](Pictures/system_design-dependency_structure.png)
+
+The project's architecture is broken down to 6 parts:
+1. Pages
+    * Serves a distinct user function, as described in section 2 - Wireframes
+    * Contains components
+2. Components
+    * Parts of a page
+    * Can contain other components
+3. Context
+    * Allows information to be transferred from one page to another without dependency pollution.
+    * e.g. if a component dependency chain looks like this: `A->B->C->D->E`, `A` and `E` can share data without going through `B,C,D`
+4. Types
+    * Local Typescript types, interfaces, templates and the like.
+    * Avoids cluttering other packages.
+5. Utils
+    * General utilities such as parsing, string formatters, etc.
+6. Styles
+    * Stylesheet files
+7. External packages
```

Comment:
we also have a hooks directory now, where the custom hook for each page is declared